core/templates/pages/exploration-editor-page/services/user-exploration-permissions.service.ts

```diff
@@ -16,35 +16,44 @@
  * @fileoverview A service that fetches and stores the permissions
  * of a user for a particular exploration.
  */
-require('domain/utilities/url-interpolation.service.ts');
-require('services/context.service.ts');
-require('services/contextual/url.service.ts');
-
-require(
-  'pages/exploration-editor-page/exploration-editor-page.constants.ajs.ts');
-
-angular.module('oppia').factory('UserExplorationPermissionsService', [
-  '$http', 'ContextService', 'UrlInterpolationService',
-  function(
-      $http, ContextService, UrlInterpolationService) {
-    var permissionsPromise = null;
-
-    return {
-      getPermissionsAsync: function() {
-        if (!permissionsPromise) {
-          var explorationPermissionsUrl = UrlInterpolationService
-            .interpolateUrl( '/createhandler/permissions/<exploration_id>', {
-              exploration_id: ContextService.getExplorationId()
-            });
-
-          permissionsPromise = $http.get(explorationPermissionsUrl).then(
-            function(response) {
-              return response.data;
-            }
-          );
-        }
-        return permissionsPromise;
-      }
-    };
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { ContextService } from 'services/context.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class UserExplorationPermissionsService {
+  permissionsPromise: any = null;
+
+  constructor(
+    private http: HttpClient,
+    private contextService: ContextService,
+    private urlInterpolationService: UrlInterpolationService,
+  ) { }
+
+  getPermissionsAsync(): any {
```

Comment:
Can we please find the exact type and erradicate `any`?

Reply:
Done